test/dest-modes.js

```diff
@@ -331,14 +332,17 @@ describe('.dest() with custom modes', function() {
       return;
     }
 
-
     var inputBase = path.join(__dirname, './fixtures');
     var inputPath = path.join(__dirname, './fixtures/wow/suchempty');
     var expectedBase = path.join(__dirname, './out-fixtures/wow');
     var expectedPath = path.join(__dirname, './out-fixtures/wow/suchempty');
     // NOTE: Darwin does not set setgid
-    var expectedDirMode = (isDarwin ? parseInt('777', 8) : parseInt('2777', 8)) & ~process.umask();
-    var expectedFileMode = parseInt('677', 8) & ~process.umask();
+    var expectedDirMode = constants.DEFAULT_DIR_MODE;
+    if (!isDarwin) {
```

Comment:
The custom mkdirp is implemented, do we still need this check?  Maybe it needs to be improved?